### ISSUE TITLE
Dials favicon are not loaded when the dial thumbnail is "custom thumbnail from URL" 

### ISSUE THREAD

**Issue by Git-GT** [fastaddons/GroupSpeedDial#92]
**Describe the bug**
When the dial thumbnail is inserted as "custom thumbnail from URL" the dial has the desired image but the favicon on the dial title is not visible

**Steps To Reproduce**
1. I imported all from Speed dial FVD, doing so, for some reason the dials thumbnail is not generated (and the favicons are not shown on top of the dials). 
All the dials show the message "create thumbnail" and NO favicons (except this, the dials work correctly). 

2. I right click on a random dial > edit dial > thumbnail 
        GroupSpeedDial shows me a panel from where I can chose how to generate the dial thumbnail (Auto-generated thumbnail, Thumbnail from public library, Custom thumbnail from file, Custom thumbnail from URL, etc. )

3.a _Custom thumbnail from file_ (works as expected)
If I select "create thumbnail from **file**: " and select an image from my pc the thumbnail is correctly displayed on the dial **AND** the favicon corresponding to the website is downloaded and showed on top of the dial

3.b _Custom thumbnail from URL_ (the thumbnail works but the dial favicon is not downloaded)
If I select "create thumbnail from **URL**: " and paste the URL related to a remote image then the thumbnail is correctly downloaded displayed on the dial **BUT** the favicon corresponding to the dial is not showed on the left of the dial title

**Expected behavior**
When the dial thumbnail is created from an URL I would expect the same behavior of when it is created from a file, the favicon should be downloaded and shown on the dial. 
Or at least it would be nice to have an option in the edit dial menu that allows to refresh the favicon so if for some reason its download failed when the dial has been created (or the favicon changed meanwhile) it is possible to update it later

**Device info:**
 - operating system: Windows 10 (64 bit)
 - Browser and version: Firefox 94.0.2 (64 bit)
 - GroupSpeedDial version: 17.0

**Comment by fastaddons** [comment]
Hello,
Thank you for the detailed report!
I have to admit, the favicons handling is pretty messy in my code. I will have to refactor it completely.

OK so I've checked the code now because I was sure there is some simple workaround but it's actually not there anymore.
All missing favicons are auto-downloaded only after synchronization, so only if you have a cloud account.
I will be releasing new version 17.1 so I will try to fix it there. For one, download missing favicon for edited dial. And probably also automatically after the import. And I will try to somehow schedule it for existing users as well.

**Comment by roussak** [comment]
It seems, the next release isn't yet reached… I faced with this behaviour, but I wasn't sure under which conditions the favicons are not updated. So, I work it around via searching the extension's storage in the Mozilla's profile, looking for a needed favicon file with an image-viewer and replacing it with a desired one. It looks like it can be worked around via the thumbnail's option temporary switching forth-and-back as well.

**Comment by Juraj-Masiar** [comment]
Regarding the original bug, I can't believe I haven't fixed it when it was reported... it must have slipped my radar somehow.
In any case, I've just finished fixing it.

The code that handles new dial creation, especially the thumbnails / favicons / dial title update, was very old and badly written (well , since these are 3 asynchronous tasks with somewhat shared resources, it's not exactly a piece of cake...).

The upcoming version 28.5 will have this finally fixed - so favicon should be always downloaded and dial name should be fetched too. 

I'm sorry it took me so long to fix this!
